File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function () {

  var RNA = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U'
  };

  this.toRna = function (dna) {
    return dna.split('').map(function(x) {return RNA[x] }).join('');
  };

};

module.exports = DnaTranscriber;

File: ruby/rna-transcription/rna_transcription.rb
class Complement
  RNA = {
    'G' => 'C',
    'C' => 'G',
    'T' => 'A',
    'A' => 'U'
  }
  def self.of_dna(dna)
    rna = dna.split('').map{|a| RNA[a]}
    return "" unless rna.compact.count == dna.length
    rna.join('')
  end
end
module BookKeeping
  VERSION = 4
end
